spacy/pipeline/legacy/entity_linker.py

```diff
@@ -54,6 +54,7 @@ def __init__(
         get_candidates: Callable[[KnowledgeBase, Span], Iterable[Candidate]],
         overwrite: bool = BACKWARD_OVERWRITE,
         scorer: Optional[Callable] = entity_linker_score,
+        abstention_threshold: Optional[float] = None,
```

Comment:
We don't need to make this change in the `v1` version. In fact, I'd argue that we want a new version (`v3`) that introduces this new parameter. Basically this ensures that config files are as cross-compat as possible: we wouldn't want a situation where somebody has a `v2` with this new threshold, which works well on one system with a new version of spaCy, and then the same config with the same `v2` breaking down on another system with an older spaCy version.

Reply:
I would definitely put this in a new version.

Reply:
What does "new version" entail exactly? A new `EntityLinker` class with the `confidence_trehold` argument + rerouting `make_entity_linker()`?

Reply:
Ah, right, this is a pipeline change, not a model class change...

In that case I guess this is like the `overwrite` changes to components when we added that across the board. I'd need to go back and look at those PRs, but I think in that case you could add it with a default that doesn't change behavior. 

Because the v1 NEL component is a little weird this is a something we'd be deciding for the first time, but my thought is that it should stay basically frozen, so there's no need to add support for a new feature to it.

Reply:
> ...so there's no need to add support for a new feature to it.

I'm good with v1 staying frozen. 

> ...but I think in that case you could add it with a default that doesn't change behavior.

Meaning adding it to `EntityLinker`, but not adding a new class?

Reply:
> Meaning adding the argument to EntityLinker, but not adding a new class?

Yes. When we added the `overwrite` option to all the components, we used a default that didn't change behavior and therefore didn't add a new architecture version. See #9050. I think we can handle this the same way.

Reply:
Yes, in the past we've added new optional parameters that preserve the existing defaults without modifying the names.

It's an issue that the factory names aren't versioned so the only thing you can really do is add new settings where the default preserves the existing behavior. It does mean that it will complain if you use the auto-filled with a previous version of spacy, but if you always auto-fill from a base config it should still work with old versions. I think this is acceptable?

Reply:
Oh sorry, I totally confused things as I was mixing up between an architecture change and a factory change - as Adriane pointed out we don't typically version the latter (`entity_linker_v1` is a weird exception but as discussed that can just be frozen)

Reply:
Can this remain as is then or are further changes necessary/recommended?

Reply:
I reviewed again yesterday, the legacy version has remained unchanged which was my main point, so this particular comment can be resolved :-)